sql/planbuilder/parse_test.go

```diff
@@ -2121,6 +2140,60 @@ Create table myTable
          ENFORCED
 `,
 		},
+		{
+			Query: "SELECT x as y FROM xy GROUP BY x HAVING AVG(-y) IS NOT NULL",
+			ExpectedPlan: `
+Project
+ ├─ columns: [xy.x:1!null as y]
+ └─ Having
+     ├─ NOT
+     │   └─ avg(-xy.y):5 IS NULL
+     └─ Project
+         ├─ columns: [avg(-xy.y):5, xy.x:1!null, xy.x:1!null as y]
+         └─ GroupBy
+             ├─ select: AVG(-xy.y), xy.x:1!null
+             ├─ group: xy.x:1!null
+             └─ Table
+                 ├─ name: xy
+                 ├─ columns: [x y z]
+                 ├─ colSet: (1-3)
+                 └─ tableId: 1
+`,
+		},
+		{
+			Query: "select x as xx from xy join uv on x = u group by xx having x = 123;",
+			ExpectedPlan: `
+Project
+ ├─ columns: [xy.x:1!null as xx]
+ └─ Having
+     ├─ Eq
+     │   ├─ xy.x:1!null
+     │   └─ 123 (tinyint)
+     └─ Project
+         ├─ columns: [xy.x:1!null, xy.x:1!null as xx]
+         └─ GroupBy
+             ├─ select: xy.x:1!null
+             ├─ group: xy.x:1!null as xx
+             └─ InnerJoin
+                 ├─ Eq
+                 │   ├─ xy.x:1!null
+                 │   └─ uv.u:4!null
+                 ├─ Table
+                 │   ├─ name: xy
+                 │   ├─ columns: [x y z]
+                 │   ├─ colSet: (1-3)
+                 │   └─ tableId: 1
+                 └─ Table
+                     ├─ name: uv
+                     ├─ columns: [u v w]
+                     ├─ colSet: (4-6)
+                     └─ tableId: 2
+`,
+		},
+		{
+			// TODO: this should fail
+			Query: "select x + 1 as xx from xy join uv on x = u group by xx having x = 123;",
```

Comment:
poking around this, if there is a HAVING but not GROUP BY does it still fail? Same issue with a WINDOW clause? Same issue with cartesian join that is just `select ... from xy,uv ...`. This just seems really weird, incremental improvements are good but more info would be nice

Reply:
`HAVING` with no `GROUP BY` should fail as well, added test for this

I think we're running into issues detailed here: https://github.com/dolthub/dolt/issues/6676

Reply:
If there's an alias that is `x` it's definition is still the projection one? Subquery aliases also. For `select x + 1 as xx, (select 1 from uv having x = 123) from xy, uv group by xx)`, I'm assuming the x is still visible inside the subquery? Lots other patterns that might be weird, HAVING subquery is another one `select x + 1 as xx from xy join uv on x = u group by xx having (select 1 from u where x = 123);`